sympy/polys/tests/test_polytools.py

```diff
@@ -2881,7 +2882,7 @@ def test_cancel():
     M = MatrixSymbol('M', 5, 5)
     assert cancel(M[0,0] + 7) == M[0,0] + 7
     expr = sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2] / z
-    assert cancel(expr) == expr
+    assert simplify(cancel(expr)) == expr
```

Comment:
yeah it returns `cancel(expr) = (z*sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2]) / z` and previously it was returning the expression `sin(M[1, 4] + 5*M[2, 1]*M[4, 0]) - 5*M[1, 2]/z`, please note that the whole numerator is encloses in a bracket after `is_commutative` is set to `True` hence I used `simplify`